web/src/components/vis/PcaPage/ScorePlot.vue

```diff
@@ -0,0 +1,93 @@
+<script>
+import ScorePlot from '@/components/vis/ScorePlot.vue';
+import VisTile from '@/components/vis/VisTile.vue';
+import plotData from '@/components/vis/mixins/plotData';
+
+export default {
+  components: {
+    ScorePlot,
+    VisTile,
+  },
+
+  mixins: [
+    plotData('pca'),
+  ],
+
+  props: {
+    id: {
+      type: String,
+      required: true,
+    },
+
+    pcX: {
+      type: Number,
+      validator: prop => Number.isInteger(prop),
+      required: true,
+    },
+
+    pcY: {
+      type: Number,
+      validator: prop => Number.isInteger(prop),
+      required: true,
+    },
+
+    showEllipses: {
+      type: Boolean,
+      required: true,
+    },
+  },
+
+  computed: {
+    ready() {
+      return this.$store.getters.ready(this.id);
+    },
+
+    pcCoords() {
+      return this.maybeData('x', []);
+    },
+
+    rowLabels() {
+      return this.maybeData('rows', []);
+    },
+
+    groupLabels() {
+      return this.maybeData('labels', {});
+    },
+
+    eigenvalues() {
+      return this.maybeData('sdev', []);
+    },
+
+    columns() {
+      return this.dataset.column.data;
+    },
+  },
+
+  methods: {
+    maybeData(key, dflt) {
+      const {
+        plot,
+      } = this;
+
+      return plot.data ? plot.data[key] : dflt;
+    },
+  },
+};
+</script>
+
+<template lang="pug">
+vis-tile(title="PCA Score Plot", :loading="plot.loading", svg-download, v-if="plot")
+  score-plot(
+      :pc-coords="pcCoords",
+      :row-labels="rowLabels",
+      :colors="dataset.groupLevels",
+      :group-labels="groupLabels",
+      :eigenvalues="eigenvalues",
+      :columns="columns",
+      :pc-x="pcX",
+      :pc-y="pcY",
+      :show-ellipses="showEllipses")
+div(v-else)
```

Comment:
I'm not sure this loader belongs at this level in the tree.  If `plot` is not defined, the dataset has yet to load, so you'll have 3 graphs with spinners that are all showing the same load event.

Might be appropriate to bump this up to the parent and not show the grid at all until the dataset is loaded.   `plot` is really just a proxy for `getters.ready`.

Reply:
I promoted this check up to the page level, using the store directly to check for data readiness. Check out my approach in 29ca24d and let me know if that looks good.